client/utils/custom-hooks.js

```diff
@@ -13,3 +13,30 @@ export const useDidUpdate = (callback, deps) => {
     }
   }, deps);
 };
+
+// Usage: const ref = useModalBehavior(() => setSomeState(false))
+// place this ref on a component
+export const useModalBehavior = (hideOverlay) => {
+  const ref = useRef({});
+
+  // Return values
+  const setRef = (r) => { ref.current = r; };
+  const [visible, setVisible] = useState(true);
+  const trigger = () => setVisible(true);
```

Comment:
I noticed that when you use <kbd>Tab</kbd> to get to the Dropdown button, and then hit <kbd>Enter</kbd>, the Dropdown will open, but if you hit <kbd>Enter</kbd> again, it won't close. Is it because `trigger` only sets `visible` to true when it should toggle it (i.e. visible = !visible)?

Reply:
You're right, thanks for catching that! I updated it and it now behaves exactly like the original dropdown lists ✨